src/Services/InAppPurchasesService.php

```diff
@@ -33,4 +37,27 @@ public function listPurchases(string $extensionName, Context $context): InAppPur
     {
         return $this->client->listInAppPurchases($extensionName, $context);
     }
+
+    /**
+     * @param Collection<covariant InAppPurchaseIdentifiableStruct> $allPurchases
+     * @param array<int, string> $validPurchases
+     *
+     * @return Collection<covariant InAppPurchaseIdentifiableStruct>
+     */
+    public function filterValidInAppPurchases(
+        Collection $allPurchases,
+        array $validPurchases,
+    ): Collection {
```

Comment:
I find this function way to complicated with the covariant stuff (and the necessary asserts). I would just do this as separate functions of each of the two collections. Then you also don't need the interface.